core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -56,6 +57,15 @@ trait TraverseFilter[F[_]] extends FunctorFilter[F] {
    */
   def traverseFilter[G[_], A, B](fa: F[A])(f: A => G[Option[B]])(implicit G: Applicative[G]): G[F[B]]
 
+  /**
+   * A combined [[traverse]] and [[collect]].
+   */
+  def traverseCollect[G[_], A, B](fa: F[A], f: PartialFunction[A, G[B]])(implicit
```

Comment:
Currying makes it easier to pass a lambda, and is consistent (e.g. see the signature above).

```suggestion
  def traverseCollect[G[_], A, B](fa: F[A])(f: PartialFunction[A, G[B]])(implicit
```